stablehlo/dialect/TypeInference.h

```diff
@@ -157,7 +157,7 @@ LogicalResult inferComplexOp(std::optional<Location> location, Value lhs,
                              SmallVectorImpl<Type>& inferredReturnTypes);
 
 LogicalResult inferConcatenateOp(std::optional<Location> location,
-                                 ValueRange inputs, int64_t dimension,
+                                 TypeRange inputTypes, int64_t dimension,
```

Comment:
I like the idea of simplifying signatures in TypeInference.h to use the least possible amount of information. Should we also have a ticket to downgrade Value/ValueRange to Type/TypeRange as much as possible in other functions in this file? cc @zhouxin913

Reply:
"as much as possible" I looked into this yesterday: I think most of them can be done. There could be very few of cases where we need the Value itself: https://github.com/openxla/stablehlo/blob/main/stablehlo/dialect/StablehloOps.cpp#L133
(I'm a little bit surprised this function is not moved)
But it can stay in `hlo_ops.cc` or we look for other ways to work around.

@sdasgup3 could you open a bug for this refactor? and add link of this PR? Thanks!

Reply:
Thanks for the insights.

Here is the ticket: https://github.com/openxla/stablehlo/issues/1028